generator/src/main/java/com/scottlogic/deg/generator/guice/RowSpecDataBagSourceFactoryProvider.java

```diff
@@ -4,27 +4,27 @@
 import com.google.inject.Provider;
 import com.scottlogic.deg.generator.generation.GenerationConfig;
 import com.scottlogic.deg.generator.generation.GenerationConfigSource;
-import com.scottlogic.deg.generator.generation.databags.ReductiveRandomRowSpecDataBagSourceFactory;
-import com.scottlogic.deg.generator.generation.databags.RowSpecDataBagSourceFactory;
-import com.scottlogic.deg.generator.generation.databags.StandardRowSpecDataBagSourceFactory;
+import com.scottlogic.deg.generator.generation.databags.ReductiveRandomRowSpecDataBagGenerator;
+import com.scottlogic.deg.generator.generation.databags.RowSpecDataBagGenerator;
+import com.scottlogic.deg.generator.generation.databags.StandardRowSpecDataBagGenerator;
 
-public class RowSpecDataBagSourceFactoryProvider implements Provider<RowSpecDataBagSourceFactory> {
+public class RowSpecDataBagSourceFactoryProvider implements Provider<RowSpecDataBagGenerator> {
     private final GenerationConfigSource configSource;
-    private final StandardRowSpecDataBagSourceFactory normalFactory;
-    private final ReductiveRandomRowSpecDataBagSourceFactory reductiveRandomFactory;
+    private final StandardRowSpecDataBagGenerator normalFactory;
+    private final ReductiveRandomRowSpecDataBagGenerator reductiveRandomFactory;
 
     @Inject
     public RowSpecDataBagSourceFactoryProvider(
         GenerationConfigSource configSource,
-        StandardRowSpecDataBagSourceFactory normalFactory,
-        ReductiveRandomRowSpecDataBagSourceFactory reductiveRandomFactory) {
+        StandardRowSpecDataBagGenerator normalFactory,
+        ReductiveRandomRowSpecDataBagGenerator reductiveRandomFactory) {
         this.configSource = configSource;
         this.normalFactory = normalFactory;
         this.reductiveRandomFactory = reductiveRandomFactory;
     }
 
     @Override
-    public RowSpecDataBagSourceFactory get() {
+    public RowSpecDataBagGenerator get() {
```

Comment:
This method contains very **similar** switching logic to that of `CombinationStrategyProvider` are we missing some indication (at least) of an improvement here?

Reply:
the real improvement here is that there eventually doesn't need to be a rowSpecDataBagGenerator in the reductive walker, so this would be able to be cleared up.

but that's a bit out of scope for this change